source/index.ts

```diff
@@ -0,0 +1,744 @@
+	}
+
+	/**
+	Get an item.
+
+	@param key - The key of the item to get.
+	@param defaultValue - The default value if the item does not exist.
+	*/
+	get(key: string, defaultValue?: T): T | undefined {
+		if (this._options.accessPropertiesByDotNotation) {
+			return dotProp.get(this.store, key, defaultValue);
+		}
+
+		return key in this.store ? this.store[key] : defaultValue;
+	}
+
+	/**
+	Set an item.
+
+	@param key - You can use [dot-notation](https://github.com/sindresorhus/dot-prop) in a key to access nested properties.
+	@param value - Must be JSON serializable. Trying to set the type `undefined`, `function`, or `symbol` will result in a `TypeError`.
+	*/
+	set(key: string | {[key: string]: T}, value?: T): void {
+		if (typeof key !== 'string' && typeof key !== 'object') {
+			throw new TypeError(`Expected \`key\` to be of type \`string\` or \`object\`, got ${typeof key}`);
+		}
+
+		if (typeof key !== 'object' && value === undefined) {
+			throw new TypeError('Use `delete()` to clear values');
+		}
+
+		if (this._containsReservedKey(key)) {
+			throw new TypeError(`Please don't use the ${INTERNAL_KEY} key, as it's used to manage this module internal operations.`);
+		}
+
+		const {store} = this;
+
+		const set = (key: string, value: T): void => {
+			checkValueType(key, value);
+			if (this._options.accessPropertiesByDotNotation) {
+				dotProp.set(store, key, value);
+			} else {
+				store[key] = value;
+			}
+		};
+
+		if (typeof key === 'object') {
+			const object = key;
+			for (const [key, value] of Object.entries(object)) {
+				set(key, value);
+			}
+		} else {
+			set(key, value as T);
+		}
+
+		this.store = store;
+	}
+
+	/**
+	Check if an item exists.
+
+	@param key - The key of the item to check.
+	*/
+	has(key: string): boolean {
+		if (this._options.accessPropertiesByDotNotation) {
+			return dotProp.has(this.store, key);
+		}
+
+		return key in this.store;
+	}
+
+	/**
+	Reset items to their default values, as defined by the `defaults` or `schema` option.
+
+	@param keys - The keys of the items to reset.
+	*/
+	reset(...keys: string[]): void {
+		for (const key of keys) {
+			if (this._defaultValues[key]) {
+				this.set(key, this._defaultValues[key]);
+			}
+		}
+	}
+
+	/**
+	Delete an item.
+
+	@param key - The key of the item to delete.
+	*/
+	delete(key: string): void {
+		const {store} = this;
+		if (this._options.accessPropertiesByDotNotation) {
+			dotProp.delete(store, key);
+		} else {
+			delete store[key];
+		}
+
+		this.store = store;
+	}
+
+	/**
+	Delete all items.
+	*/
+	clear(): void{
+		this.store = this.createPlainObject();
+	}
+
+	/**
+	Watches the given `key`, calling `callback` on any changes.
+
+	@param key - The key wo watch.
+	@param callback - A callback function that is called on any changes. When a `key` is first set `oldValue` will be `undefined`, and when a key is deleted `newValue` will be `undefined`.
+	*/
+	onDidChange(key: string, callback: (...args: unknown[]) => void): () => unknown {
```

Comment:
This should be strongly-typed.

https://github.com/sindresorhus/conf/blob/97bcc918a109ba964f7ef8a091cc5a762ae03576/index.d.ts#L316-L319